### PR TITLE
Edgeboxes

### PR DESCRIPTION
**Pull request by lhelontra**
This is an edgeboxes algorithm ported from [pdollar](https://github.com/pdollar/edges/). 
This implementation uses edges image and orientation map, provides by StructuredEdgeDetection, already implemented.

### CODE REVIEW

**Comment by StevenPuttemans** [comment]
Hi @lhelontra , does this also include the edge box region proposal method? This might be interesting to use as a region proposal system for the deep learning modules.

**Comment by lhelontra** [comment]
Hi @StevenPuttemans , this implementation have a getBoundingBoxes function, returns bounding boxes proposals.

**Comment by sovrasov** [comment]
@lhelontra the implementation looks great. Could you add some tests or at least a sample which uses this algorithm?

File: modules/ximgproc/include/opencv2/ximgproc/edgeboxes.hpp [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+// loss of use, data, or profits; or business interruption) however caused
+// and on any theory of liability, whether in contract, strict liability,
+// or tort (including negligence or otherwise) arising in any way out of
+// the use of this software, even if advised of the possibility of such damage.
+//
+//M*/
+
+#ifndef __OPENCV_EDGEBOXES_HPP__
+#define __OPENCV_EDGEBOXES_HPP__
+#ifdef __cplusplus
```
Comment (sovrasov) [inline modules/ximgproc/include/opencv2/ximgproc/edgeboxes.hpp]: This protection seems to be useless.

Comment (lhelontra) [inline modules/ximgproc/include/opencv2/ximgproc/edgeboxes.hpp]: @sovrasov I added two samples, in python and c++. About this protection, I followed the style of structuredEdgeDetection header, you suggest remove this?

Comment (sovrasov) [inline modules/ximgproc/include/opencv2/ximgproc/edgeboxes.hpp]: Yes, this style comes from the old times when OpenCV was a pure C library.

Comment (lhelontra) [inline modules/ximgproc/include/opencv2/ximgproc/edgeboxes.hpp]: Thanks for the info. I will keep this in mind for future contributions.

**Comment by lhelontra** [comment]
Oh sure, my mistake.

**Comment by sovrasov** [comment]
👍 but squash your commits into one before merge

**Comment by ThorbjornSomod** [comment]
@lhelontra I am very eager to try out the region proposals of the edgebox algorithm in my implementation. I therefore uninstalled my version of OpenCV, added your commits and rebuilt OpenCV. However when I try to run your edgeboxes_demo.py with a test image and the edge model provided at opencv_extra/testdata/cv/ximgproc/model.yml.gz I get the error:

Traceback (most recent call last):
    File "edgeboxes_demo.py", line 24, in <module>
        orimap = edge_detection.computeOrientation(edges)
AttributeError: 'cv2.ximgproc_StructuredEdgeDetection' object has no attribute 'computeOrientation'

Anything else I should change than what is in the commit log?

**Comment by sturkmen72** [comment]
is this PR ready to merge?

**Comment by lhelontra** [comment]
@sturkmen72 yes

**Comment by zchrissirhcz** [comment]
@lhelontra  I'd like to know, what is the structured random forest model file should be like? I use the demo python code, and use Piotr Dollar's original model (`edges/models/forest/modelBsds.mat`), but fails. It cannot parse that matlab model. How can I get a such proper model? Thanks.

**Comment by zchrissirhcz** [comment]
@lhelontra Sorry for disturb. I find the convert scripts in `opencv_contrib/modules/ximgproc/tutorials/scripts`.

**Comment by lhelontra** [comment]
@zchrissirhcz no problem, you can uses this model: https://github.com/opencv/opencv_extra/blob/master/testdata/cv/ximgproc/model.yml.gz

**Comment by zchrissirhcz** [comment]
@lhelontra  Thanks, got that.

**Pull request merged**